Realm/RLMResults.mm

```diff
@@ -376,6 +376,9 @@ - (id)objectAtIndexedSubscript:(NSUInteger)index {
 }
 
 - (id)aggregate:(NSString *)property method:(util::Optional<Mixed> (Results::*)(size_t))method methodName:(NSString *)methodName {
+    if (_results.get_mode() == Results::Mode::Empty) {
```

Comment:
You're right that sum should be returning zero in this case. It looks like `Results::aggregate`, which is what I looked at when checking what the behavior should be, also gets this wrong for the `Mode::Empty` case. It is not respecting the `return_none_for_empty` argument as it should be.